vmdb/app/models/miq_host_provision_workflow.rb

```diff
@@ -75,12 +75,15 @@ def allowed_hosts(options={})
   end
 
   def allowed_ws_hosts(options={})
-    Host.find(:all, :conditions=>["mac_address is not NULL"]).find_all {|h| h.ipmi_enabled}
+    Host.where("mac_address is not NULL").select { |h| h.ipmi_enabled }
```

Comment:
Could this be query-ized to avoid pulling everything back?

Something like `Host.where("mac_address is not NULL").where(:ipmi_enabled => true)`

Although this might fall into the Arel bug I was talking about earlier.